src/util/RadialBarUtils.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { isValidElement, cloneElement, SVGProps } from 'react';
+import _ from 'lodash';
+import { RadialBarProps } from '../polar/RadialBar';
+import { Sector, Props as SectorProps } from '../shape/Sector';
+import { Layer } from '../container/Layer';
+
+// Sector props is expecting cx, cy as numbers.
+// When props are being spread in from a user defined component in RadialBar,
+// the prop types of an SVGElement have these typed as string | number.
+// This function will return the passed in props along with cx, cy as numbers.
+export function typeGuardSectorProps(option: SVGProps<SVGPathElement>, props: SectorProps) {
+  const cxValue = `${props.cx || option.cx}`;
+  const cx = parseInt(cxValue, 10);
+  const cyValue = `${props.cy || option.cy}`;
+  const cy = parseInt(cyValue, 10);
+  return {
+    ...props,
+    ...option,
+    cx,
+    cy,
+  };
+}
+
+export interface RadialBarSectorProps extends SectorProps {
+  index?: number;
+  option: RadialBarProps['activeShape'];
+  isActive: boolean;
+}
+
+export function RadialBarSector({ option, ...props }: RadialBarSectorProps) {
+  let sectorShape: React.JSX.Element;
+
+  if (isValidElement(option)) {
+    sectorShape = cloneElement(option, props);
+  } else if (_.isFunction(option)) {
+    sectorShape = option(props);
+  } else if (_.isPlainObject(option) && !_.isBoolean(option)) {
+    const elementProps = typeGuardSectorProps(option, props);
+    sectorShape = <Sector {...elementProps} />;
+  } else {
+    sectorShape = <Sector {...props} />;
+  }
```

Comment:
I have seen this logic before in various variants. Should we possibly extract this as a helper and reuse? 

<img width="860" alt="Screenshot 2023-10-01 at 12 14 46" src="https://github.com/recharts/recharts/assets/27995606/196877be-4633-4ce9-b7eb-7e9b4b59d289">

Reply:
Addressed here: https://github.com/recharts/recharts/pull/3803/commits/d28ad574c21165a70e11697e70a9e753a7d063c3

Let me know if you have any feedback on that refactor.